src/state/ledger/mod.rs

```diff
@@ -302,4 +302,65 @@ mod tests {
 
         assert_eq!(account_after.delegate, Some(delegate_key));
     }
+
+    #[test]
+    fn apply_diff_payment_with_post_balance() {
+        let public_key =
+            PublicKey::from_address("B62qre3erTHfzQckNuibViWQGyyKwZseztqrjPZBv6SQF384Rg6ESAy")
+                .expect("public key creation");
+        let mut account = Account::empty(public_key.clone());
+        account.balance = Amount(10); // Set the balance explicitly
+        let mut accounts = HashMap::new();
+        accounts.insert(public_key.clone(), account);
+        let mut ledger = Ledger { accounts };
+
+        let ledger_diff = LedgerDiff {
+            public_keys_seen: vec![],
+            account_diffs: vec![AccountDiff::Payment(PaymentDiff {
+                public_key: public_key.clone(),
+                amount: 1, // Pass the underlying value
+                update_type: UpdateType::Deposit,
+            })],
+        };
+
+        ledger
+            .apply_diff(&ledger_diff)
+            .expect("ledger diff application");
+
+        let account_after = ledger.accounts.get(&public_key).expect("account get");
+
+        assert_eq!(account_after.balance, Amount(11)); // Assert against the balance
```

Comment:
Can you implement an `add(&self, amt: uint64)` method on `Amount` in `src/state/ledger/account.rs` and explicitly write the `Amount(11)` value as `initial_balance.add(ledger_diff.account_diffs[0].amount)`?